lib/python/rose/macros/value.py

```diff
@@ -96,7 +96,7 @@ def _validate_id(self, var_id, value, meta_config):
         sect, key = self._get_section_option_from_id(var_id)
         saved_metadata = copy.deepcopy(metadata)
         saved_metadata.pop('id')
-        for meta_key in saved_metadata.keys():
+        for meta_key in list(saved_metadata.keys()):
```

Comment:
Yes, since `saved_metadata.pop()` is used inside the loop.  
This is what would happen..
```python3
>>> _dict_= {'a': 1, 'b': 2}
>>> for item in _dict_.keys():
...  _dict_.pop(item)
... 
1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: dictionary changed size during iteration
```

Reply:
👍